proxy/proxy_test.go

```diff
@@ -2444,12 +2444,12 @@ var _ = Describe("Proxy", func() {
 					req, err := http.ReadRequest(conn.Reader)
 					Expect(err).NotTo(HaveOccurred())
 
-					done <- req.Header.Get("Upgrade") == "WebsockeT" &&
-						req.Header.Get("Connection") == "UpgradE"
+					done <- req.Header.Get("Upgrade") == "Websocket" &&
+						req.Header.Get("Connection") == "Upgrade"
```

Comment:
Why did you adjust these tests?

Reply:
The httputil.ReverseProxy takes care of sanitizing the hop-by-hop headers, too. After removing them according to [RFC 7230](https://github.com/golang/go/blob/master/src/net/http/httputil/reverseproxy.go#L294) it re-adds the headers necessary for the websocket update [here](https://github.com/golang/go/blob/master/src/net/http/httputil/reverseproxy.go#L413) but it adds them like a sane person would, so "UpdgraE" sent by your client becomes "Upgrade" at the other end.

The tests adjusted make the wrong assumptions that these headers are retained as-is and are not touched by the proxy - even though they are hop-by-hop headers that are supposed to be consumed by proxies.